src/python/genny/genny_auto_tasks.py

```diff
@@ -113,36 +226,58 @@ def construct_task_json(workloads, variants):
 
 def main():
     """
-    Main Function: parses args, writes to file evergreen tasks in json format for workloads that have been modified locally.
+    Main Function: parses args, writes to file evergreen tasks in json format for workloads that are specified in args.
     """
     parser = argparse.ArgumentParser(
         description="Generates json that can be used as input to evergreen's generate.tasks, representing genny workloads to be run")
 
     parser.add_argument('--variants', nargs='+', required=True, help='buildvariants that workloads should run on')
-    parser.add_argument('--workloads', nargs='+',
-                        help='paths of workloads to run, relative to src/workloads/ in the genny repository root')
     parser.add_argument('-o', '--output', default='build/generated_tasks.json',
                         help='path of file to output result json to, relative to the genny root directory')
+
+    workload_group = parser.add_mutually_exclusive_group(required=True)
+    workload_group.add_argument('--autorun', action='store_true', default=False,
+                                help='if set, the script will generate tasks based on workloads\' AutoRun section and bootstrap.yml/runtime.yml files in the working directory')
+    workload_group.add_argument('--modified', action='store_true', default=False,
+                                help='if set, the script will generate tasks for workloads that have been added/modifed locally, relative to origin/master')
+    workload_group.add_argument('--workloads', nargs='+',
+                                help='paths of workloads to run, relative to src/workloads/ in the genny repository root')
+
     args = parser.parse_args(sys.argv[1:])
+    original_cwd = os.getcwd()
+    cd_genny_root()
 
-    if args.workloads is not None:
-        errs = validate_user_workloads(args.workloads)
-        if len(errs) > 0:
-            for e in errs:
-                print('invalid workload: {}'.format(err), file=sys.stderr)
+    if args.autorun:
+        env_dict = make_env_dict(original_cwd)
+        if env_dict is None:
+            print(
+                'fatal error: bootstrap.yml and runtime.yml files not found in current directory, cannot AutoRun workloads\n\
+                note: --autorun is intended to be called from within Evergreen. If using genny locally, please run the workload directly.',
+                file=sys.stderr)
+            print(os.getcwd(), file=sys.stderr)
             return
-        workloads = args.workloads
-    else:
+
+        workloads = autorun_workload_files(env_dict)
+        if len(workloads) == 0:
+            print('No AutoRun workloads found matching environment, generating no tasks.')
```

Comment:
Hm, could we throw an error in this case? I think one use case is for people adding new sys-perf build variants. They will want to run Genny workloads on them, but they may not know to change workload YAMLs to support the new build variant. Errorring and telling people to double check the workload YAML files could be a safeguard, though it's not foul-proof.

Reply:
Done. I did the same if no workloads are generated with the `--modified` flag because I imagine that someone would only use it if they expected it to spawn some workloads.